service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java

```diff
@@ -99,64 +96,132 @@ public void setRestTemplate(RestTemplate restTemplate) {
+                errorResponse.addProperty("message", "Invalid JSON format");
+                return new ResponseEntity<>(errorResponse, HttpStatus.BAD_REQUEST);
+            }
+        } catch (Exception e) {
+            log.error("Unexpected exception caught", e);
+            errorResponse.addProperty("Status code", HttpStatus.BAD_REQUEST.value());
+            errorResponse.addProperty("Result", "Fail");
+            errorResponse.addProperty("Message", "Invalid JSON format");
+            return new ResponseEntity<>(errorResponse, HttpStatus.INTERNAL_SERVER_ERROR);
+
+        }
+    }
+
+    /**
+     * This helper method basically generate or process one event
+     * @return JsonObject generated event
+     */
+
+    public JsonObject processEvent(JsonObject jsonElement, String msgProtocol, String msgType, Boolean failIfMultipleFound,
+                                   Boolean okToLeaveOutInvalidOptionalFields,
+                                   Boolean failIfNoneFound, int lookupLimit, Boolean lookupInExternalERs) {
+        JsonObject eventResponse = new JsonObject();
+        JsonElement parsedResponse = null;
+        try {
+            JsonObject event = jsonElement.getAsJsonObject();
+            event = erLookup(event, failIfMultipleFound, failIfNoneFound, lookupInExternalERs, lookupLimit);
             MsgService msgService = getMessageService(msgProtocol);
-            String response;
+
             if (msgService != null) {
-                response = msgService.generateMsg(msgType, bodyJson, isLenientEnabled(okToLeaveOutInvalidOptionalFields));
-                JsonElement parsedResponse = parser.parse(response);
-                if(lookupLimit <= 0) {
-                    return new ResponseEntity<>("LookupLimit must be greater than or equals to 1", HttpStatus.BAD_REQUEST);
-                }
-                if (!parsedResponse.getAsJsonObject().has(RemremGenerateServiceConstants.JSON_ERROR_MESSAGE_FIELD)) {
-                    return new ResponseEntity<>(parsedResponse, HttpStatus.OK);
+                String response = msgService.generateMsg(msgType, event, isLenientEnabled(okToLeaveOutInvalidOptionalFields));
+                parsedResponse = parser.parse(response);
+
+                if (lookupLimit <= 0) {
+                    eventResponse.addProperty("status code", HttpStatus.BAD_REQUEST.value());
+                } else if (!parsedResponse.getAsJsonObject().has(RemremGenerateServiceConstants.JSON_ERROR_MESSAGE_FIELD)) {
+                    return parsedResponse.getAsJsonObject();
                 } else {
-                    return new ResponseEntity<>(parsedResponse, HttpStatus.BAD_REQUEST);
+                    eventResponse.addProperty("Status code", HttpStatus.BAD_REQUEST.value());
+                    eventResponse.addProperty("Result", "Fail");
+                    eventResponse.addProperty("Message", RemremGenerateServiceConstants.NOT_ACCEPTABLE);
```

Comment:
Can we understand what went wrong reading the error message?

Reply:
Added extra template error constant variable in RemremGenerateServiceConstants.class